category_theory/types.lean

```diff
@@ -40,10 +40,27 @@ variables {D : Type u'} [𝒟 : category.{u' v'} D] (I J : D ⥤ C) (ρ : I ⟹
 
 end functor_to_types
 
+def ulift_trivial (V : Type u) : ulift.{u} V ≅ V := by tidy
+
 def ulift_functor : (Type u) ⥤ (Type (max u v)) :=
 { obj := λ X, ulift.{v} X,
   map := λ X Y f, λ x : ulift.{v} X, ulift.up (f x.down) }
 
+@[simp] lemma ulift_functor.map {X Y : Type u} (f : X ⟶ Y) (x : ulift.{v} X) :
+  ulift_functor.map f x = ulift.up (f x.down)
+:= rfl
+
+instance ulift_functor_faithful : faithful ulift_functor :=
```

Comment:
Isn't it better for the type class system if we prove `fully_faithful`?

Reply:
Done. (And removed `tidy` from the proof...)